[RenderStyleGen] Extend getter/setter generation to all non-special cased properties

weinig opened the pull request:
#### b59b7d45815ea9c0532af8ae529053c61da1abe4
<pre>
[RenderStyleGen] Extend getter/setter generation to all non-special cased properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=303151">https://bugs.webkit.org/show_bug.cgi?id=303151</a>

Reviewed by Darin Adler.

Extends RenderStyle getter/setter function implementation generation
to all properties except ones that still need additional special casing.

The remaining special cases are:
    - properties that use fromRaw/toRaw conversions (e.g. `hanging-punctuation`)
    - properties that have setters that set multiple values (e.g. `appearance`)
    - properties that don&apos;t have a single strong style type representing them (e.g. `caret-color`)
    - properties that have a getter name that doesn&apos;t match the setter name (e.g. `computedWordSpacing` / `setWordSpacing`)
    - properties that are not stored directly in a data class, but rather within a struct (e.g. `border-bottom-color`)
    - @page descriptors (e.g. `page-size`)

A few changes were need to conform all the properties that are being generated
including:
    - update the names of storage members to match the property name.
    - adding a data class for `backdrop-filter` to ensure we can have a matching storage member name.
    - converting `Style::ScrollbarWidth` from a struct to an enum, and using a ToPlatform interface
      to implement the platform conversion.
    - Updating the signatures of getters/setters in RenderStyle to match conventions

Additionally, a few member functions of RenderStyle were easily removable:
    - aspectRatioWidth()/aspectRatioHeight() were replaced by aspectRatio().width()/aspectRatio().height()
    - clearAnimations()/clearTransitions() were unused.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/scripts/process-css-properties.py:
* Source/WebCore/page/LocalFrameView.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebCore/rendering/RenderTextControl.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/style/StyleBackdropFilterData.cpp: Added.
* Source/WebCore/rendering/style/StyleBackdropFilterData.h: Added.
* Source/WebCore/rendering/style/StyleBackgroundData.cpp:
* Source/WebCore/rendering/style/StyleBackgroundData.h:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleBoxData.h:
* Source/WebCore/rendering/style/StyleDeprecatedFlexibleBoxData.cpp:
* Source/WebCore/rendering/style/StyleDeprecatedFlexibleBoxData.h:
* Source/WebCore/rendering/style/StyleGridData.cpp:
* Source/WebCore/rendering/style/StyleGridData.h:
* Source/WebCore/rendering/style/StyleGridItemData.cpp:
* Source/WebCore/rendering/style/StyleGridItemData.h:
* Source/WebCore/rendering/style/StyleMarqueeData.cpp:
* Source/WebCore/rendering/style/StyleMarqueeData.h:
* Source/WebCore/rendering/style/StyleMultiColData.cpp:
* Source/WebCore/rendering/style/StyleMultiColData.h:
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareInheritedData.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/style/StyleTransformData.cpp:
* Source/WebCore/rendering/style/StyleTransformData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/values/scrollbars/StyleScrollbarWidth.cpp:
* Source/WebCore/style/values/scrollbars/StyleScrollbarWidth.h:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:

Canonical link: <a href="https://commits.webkit.org/303585@main">https://commits.webkit.org/303585@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8216652e309cde1b8307df7465861fbd26d87b5b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132906 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5407 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43997 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140441 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84937 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134776 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5798 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5270 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101627 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68946 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135852 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/4073 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119078 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82426 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3961 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1586 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83674 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112982 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37196 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143093 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5076 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37779 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110003 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5158 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4345 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110184 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27934 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3890 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115342 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58627 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5130 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/33696 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4970 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68582 "Built successfully") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5220 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5088 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/rendering/style/StyleBackdropFilterData.h:
```diff
@@ -0,0 +1,57 @@
+
+namespace WTF {
+class TextStream;
+}
```
How does it help to have this forward decoration without the `using?`

weinig replied:
It's a copy paste job. It's not needed at all. I'll fix in both places.

darinadler commented on Source/WebCore/rendering/style/RenderStyleSetters.h:
```diff
@@ -589,6 +224,19 @@ inline void RenderStyle::setLogicalMaxHeight(Style::MaximumSize&& height)
+
+// FIXME: - Below are property setters that are not yet generated
+
+// FIXME: Support setters than need to return a `bool` value to indicate if the property changed.
```
```suggestion
// FIXME: Support setters that need to return a `bool` value to indicate if the property changed.
```

weinig committed:
[RenderStyleGen] Extend getter/setter generation to all non-special cased properties
https://bugs.webkit.org/show_bug.cgi?id=303151

Reviewed by Darin Adler.

Extends RenderStyle getter/setter function implementation generation
to all properties except ones that still need additional special casing.

The remaining special cases are:
    - properties that use fromRaw/toRaw conversions (e.g. `hanging-punctuation`)
    - properties that have setters that set multiple values (e.g. `appearance`)
    - properties that don't have a single strong style type representing them (e.g. `caret-color`)
    - properties that have a getter name that doesn't match the setter name (e.g. `computedWordSpacing` / `setWordSpacing`)
    - properties that are not stored directly in a data class, but rather within a struct (e.g. `border-bottom-color`)
    - @page descriptors (e.g. `page-size`)

A few changes were need to conform all the properties that are being generated
including:
    - update the names of storage members to match the property name.
    - adding a data class for `backdrop-filter` to ensure we can have a matching storage member name.
    - converting `Style::ScrollbarWidth` from a struct to an enum, and using a ToPlatform interface
      to implement the platform conversion.
    - Updating the signatures of getters/setters in RenderStyle to match conventions

Additionally, a few member functions of RenderStyle were easily removable:
    - aspectRatioWidth()/aspectRatioHeight() were replaced by aspectRatio().width()/aspectRatio().height()
    - clearAnimations()/clearTransitions() were unused.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/scripts/process-css-properties.py:
* Source/WebCore/page/LocalFrameView.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebCore/rendering/RenderTextControl.cpp:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/SVGRenderStyle.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/style/StyleBackdropFilterData.cpp: Added.
* Source/WebCore/rendering/style/StyleBackdropFilterData.h: Added.
* Source/WebCore/rendering/style/StyleBackgroundData.cpp:
* Source/WebCore/rendering/style/StyleBackgroundData.h:
* Source/WebCore/rendering/style/StyleBoxData.cpp:
* Source/WebCore/rendering/style/StyleBoxData.h:
* Source/WebCore/rendering/style/StyleDeprecatedFlexibleBoxData.cpp:
* Source/WebCore/rendering/style/StyleDeprecatedFlexibleBoxData.h:
* Source/WebCore/rendering/style/StyleGridData.cpp:
* Source/WebCore/rendering/style/StyleGridData.h:
* Source/WebCore/rendering/style/StyleGridItemData.cpp:
* Source/WebCore/rendering/style/StyleGridItemData.h:
* Source/WebCore/rendering/style/StyleMarqueeData.cpp:
* Source/WebCore/rendering/style/StyleMarqueeData.h:
* Source/WebCore/rendering/style/StyleMultiColData.cpp:
* Source/WebCore/rendering/style/StyleMultiColData.h:
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareInheritedData.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/style/StyleTransformData.cpp:
* Source/WebCore/rendering/style/StyleTransformData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/values/scrollbars/StyleScrollbarWidth.cpp:
* Source/WebCore/style/values/scrollbars/StyleScrollbarWidth.h:
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:

Canonical link: https://commits.webkit.org/303585@main
